core/core.schema.go

```diff
@@ -88,76 +87,68 @@ func (r *coreSchema) Dependencies() []router.ExecutableSchema {
 	return nil
 }
 
-func (r *coreSchema) core(p graphql.ResolveParams) (any, error) {
-	return struct{}{}, nil
+func (r *coreSchema) image(ctx *router.Context, parent any, args struct {
+	Ref string
+}) (*filesystem.Filesystem, error) {
+	st := llb.Image(args.Ref, llb.WithMetaResolver(r.gw))
+	return r.Solve(ctx, st)
 }
 
-func (r *coreSchema) host(p graphql.ResolveParams) (any, error) {
-	return struct{}{}, nil
-}
-
-func (r *coreSchema) image(p graphql.ResolveParams) (any, error) {
-	ref := p.Args["ref"].(string)
-
-	st := llb.Image(ref, llb.WithMetaResolver(r.gw))
-	return r.Solve(p.Context, st)
-}
-
-func (r *coreSchema) git(p graphql.ResolveParams) (any, error) {
-	remote := p.Args["remote"].(string)
-	ref, _ := p.Args["ref"].(string)
-
+func (r *coreSchema) git(ctx *router.Context, parent any, args struct {
```

Comment:
How about a grpc/rest like convention:

```go
type gitRequest {
  stuff
}

type gitResponse {
  stuff
}

func git(args gitRequest) (gitResponse, error) {
 ...
}
```

rather than inlining the args struct

- of course stuff that returns a type directly (e.g. git returning Filesystem) doesn't need a response time
- `*Request/*Response` struct names were just examples. Maybe `*Args` or something else is more apt
- would it be possible to return a type rather than any (have the wrapper func deal with that)?

Reply:
Updated to use `*Args`

> would it be possible to return a type rather than any (have the wrapper func deal with that)?

Yep that was already handled, I just forgot to change some of the return args to be not `any`. Think I got them all now.